src/Terrain/TerrainType.java

```diff
@@ -158,6 +160,11 @@ private TerrainType(int attributeFlags, int defense, String name, TerrainType ba
   private static final String ROAD_NAME = "ROAD";
   public static final TerrainType ROAD = new TerrainType( ROAD_FLAGS, ROAD_DEFENSE, ROAD_NAME, SHOAL );
 
+  private static final int TELETILE_FLAGS = UNWEATHERABLE;
+  private static final int TELETILE_DEFENSE = 0;
+  private static final String TELETILE_NAME = "TELETILE";
+  public static final TerrainType TELETILE = new TerrainType( TELETILE_FLAGS, TELETILE_DEFENSE, TELETILE_NAME, null );
+
```

Comment:
Excellent. Mag tubes are definitely what we were still missing. Except these are better, since they work on air and sea units as well.

All we need now is a sample map that contains them.

Reply:
By your command.

Reply:
It is... Glorious. Better than I could have imagined.